nvchecker/get_version.py

```diff
@@ -13,13 +14,45 @@
   'anitya',
 )
 
+def substitute_version(version, name, conf):
+  '''
+  Substitute the version string via defined rules in the configuration file.
+  Accepted rules are:
+
+  * from_pattern, to_pattern: Both should be Python regular expressions.
+    `from_pattern` found in the version string will be substituted to
+    `to_pattern`.
+  * prefix: If the version string starts with `prefix`, the prefix is removed.
+    Otherwise the version string is returned as-is.
+
+  If both prefix and from_pattern/to_pattern are used, from_pattern/to_pattern
+  are ignored.
+  '''
+  prefix = conf.get('prefix')
+  if prefix:
+    if version.startswith(prefix):
+      version = version[len(prefix):]
+    return version
```

Comment:
What does the user want when both `prefix` and `*_pattern` are given?

Reply:
I don't think it's reasonable to use both in a package. If stripping a prefix is what they need, they should use ```prefix``` only. If they want some complicated transformations, they should use ```*_pattern```. ```*_pattern``` should also be used when they want to strip the prefix and then do something special. For example, the transformation of ```v1_1_0``` => ```1.1.0``` can be achieved with ```from_pattern = v(\d+)_(\d+)_(\d+)``` and ```to_pattern = \1.\2.\3```.

I'll fix other issues soon. Most likely tomorrow.

Reply:
Then you should make it clear at least in the docs.

Reply:
> I don't think it's reasonable to use both in a package.
I disagree :upside_down_face: :
https://github.com/lilydjwg/nvchecker/issues/249